mat/matrix.go

```diff
@@ -749,6 +748,11 @@ func Min(a Matrix) float64 {
 	}
 }
 
+// A Normer can compute a norm of the matrix.
+type Normer interface {
+	Norm(norm float64) float64
```

Comment:
Note that valid values of `norm` here?

Reply:
I left it loose because I was thinking if we want to allow Normers to deviate from the valid values, for example a potential SVD.Norm(2) could return the spectral norm instead of the Frobenius norm.

Reply:
Fair enough.